pipeline/data_page/get_all_data_availability.py

```diff
@@ -357,37 +358,103 @@ def get_paralogs_summary(tc, depmap_paralogs_taiga_id):
     return paralogs_summary
 
 
-def get_id(possible_id, id_key="dataset_id"):
+def get_long_reads_summary(gcloud_storage_client, depmap_long_reads_gcloud_loc):
+    """
+    Get a summary of the long reads data available for each cell line.
+
+    Args:
+        gcloud_storage_client (storage.Client): gcloud storage client
+        depmap_long_reads_gcloud_loc (dict): gcloud location of the long reads data
+
+    Returns:
+        pd.DataFrame: Indexed by ACHID or ModelID, with a column of True values that will be filterd against the Model table in future
+    """
+
+    print("getting long_reads_summary...")
+
+    bucket_name = depmap_long_reads_gcloud_loc["bucket_name"]
+    prefix = depmap_long_reads_gcloud_loc["prefix"]
+    file_names = depmap_long_reads_gcloud_loc["file_names"].replace(" ", "").split(",")
+
+    # We have at least one file to process
+    assert len(file_names) > 0, "No file names provided in gcloud location"
+
+    bucket = gcloud_storage_client.bucket(bucket_name)
+    unique_model_ids = set()
+
+    for file_name in file_names:
+        blob_name = f"{prefix}/{file_name}"
+        bucket = gcloud_storage_client.bucket(bucket_name)
+        blob = bucket.blob(blob_name)
+        content = blob.download_as_string()
+        df = pd.read_csv(pd.io.common.BytesIO(content), usecols=[0])
+        # Currently, the column name is "ACHID". Could be changed in the future to "ModelID" or something else?
+        assert "ACHID" in df.columns, f"Column 'ACHID' not found in file {file_name}"
+        unique_model_ids.update(df["ACHID"].unique())
+
+    assert len(unique_model_ids) > 0, "No model IDs found in the provided files"
+    print(f"Length of unique_model_ids: {len(unique_model_ids)}")
+
+    # Note: Converting unique_model_ids to a list due to a breaking change between pandas 1.4.3 and 1.5.3 regarding using sets as DataFrame indices.
+    # Since pandas 1.5.3, using a set as an index raises a ValueError.
+    long_reads_summary = pd.DataFrame(
+        index=list(unique_model_ids), columns=["Sequencing_Long_Reads"], data=True
+    )
+
+    assert len(long_reads_summary) == len(
+        unique_model_ids
+    ), "Number of rows in DataFrame doesn't match number of unique model IDs"
+
+    return long_reads_summary
+
+
+def get_taiga_id(possible_id, id_key="dataset_id"):
     return [] if len(possible_id) == 0 else [possible_id[0][id_key]]
 
 
+def get_gcloud_loc(possible_id):
+    if len(possible_id) == 0:
+        return []
```

Comment:
This is a little unusual. I see that you're testing 
```
    if len(depmap_long_reads_gcloud_loc) > 0:
```
which is technically going to work but it's very werid that depmap_long_reads_gcloud_loc is either type of list *or* a dict, and if it's a list, it's always an empty list.

How about changing this to `None` instead? (and the corresponding test to check for `...is not None`)

As written, it looked like a mistake because it's uncommon to have mixed return types like that.